src/installer/managed/Microsoft.NET.HostModel/ResourceUpdater.cs

```diff
@@ -183,9 +185,13 @@ public static bool IsSupportedOS()
             {
                 // On Nano Server 1709+, `BeginUpdateResource` is exported but returns a null handle with a zero error
                 // Try to call `BeginUpdateResource` with an invalid parameter; the error should be non-zero if supported
+                // On Nano Server 20213, `BeginUpdateResource` fails with ERROR_CALL_NOT_IMPLEMENTED
                 using (var handle = Kernel32.BeginUpdateResource("", false))
                 {
-                    if (handle.IsInvalid && Marshal.GetLastWin32Error() == 0)
+                    int lastWin32Error = Marshal.GetLastWin32Error();
+
+                    if ((handle.IsInvalid && lastWin32Error == 0) ||
```

Comment:
```suggestion
                    if (handle.IsInvalid && (lastWin32Error == 0 || lastWin32Error == Kernel32.ERROR_CALL_NOT_IMPLEMENTED))
```
The typical contract for Win32 APIs is that last error is guaranteed to be set only when there is an actual error.